projects/components/src/table/cells/table-cell-csv-generator-management.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable, Type } from '@angular/core';
+import { TableCellCsvGeneratorBase } from './table-cell-csv-generator-base';
+import { CoreTableCellCsvGeneratorType } from './types/core-table-cell-csv-generator-type';
+import { includes, isArray } from 'lodash-es';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TableCellCsvGeneratorManagementService {
+  private readonly csvGenerators: TableCellCsvGeneratorBase<unknown>[] = [];
+
+  public register(csvGenerators: TableCellCsvGeneratorBase<unknown> | TableCellCsvGeneratorBase<unknown>[]): void {
+    isArray(csvGenerators) ? csvGenerators.forEach(item => this.addGenerator(item)) : this.addGenerator(csvGenerators);
+  }
+
+  private addGenerator(csvGenerator: TableCellCsvGeneratorBase<unknown>): void {
+    if (!includes(this.csvGenerators, csvGenerator)) {
+      this.csvGenerators.push(csvGenerator);
+    }
+  }
+
+  public findMatchingGenerator(type?: string): TableCellCsvGeneratorBase<unknown> | undefined {
+    return this.csvGenerators.find(generator => generator.cellType === type);
+  }
+}
+
+/*
+ * TableCellRendererConstructor is used by lookup service to dynamically instantiate cell CsvGenerators
+ */
+export interface TableCellCsvGeneratorConstructor<TCellData, TRowValue>
```

Comment:
do we still use decorators/constructors?

Reply:
nope. removed.